packages/quantic/force-app/main/default/lwc/quanticResultCopyToClipboard/__tests__/quanticResultCopyToClipboard.test.js

```diff
@@ -0,0 +1,301 @@
+          const element = createTestComponent();
+          await flushPromises();
+
+          const resultAction = element.shadowRoot.querySelector(
+            selectors.quanticResultAction
+          );
+          expect(resultAction).not.toBeNull();
+
+          element.dispatchEvent(copyToClipboardEvent);
+          await flushPromises();
+
+          expect(functionMocks.logCopyToClipboard).toHaveBeenCalledTimes(1);
+          expect(functionMocks.logCopyToClipboard).toHaveBeenCalledWith(
+            defaultOptions.result
+          );
+        });
+
+        it('should call the copyToClipboard function with the correct parameters when the promise resolved', async () => {
+          const mockCopyToClipboard = quanticUtils.copyToClipboard;
+          const element = createTestComponent();
+          const resultAction = element.shadowRoot.querySelector(
+            selectors.quanticResultAction
+          );
+
+          element.dispatchEvent(copyToClipboardEvent);
+          await flushPromises();
+
+          expect(mockCopyToClipboard).toHaveBeenCalledTimes(1);
+          expect(mockCopyToClipboard).toHaveBeenCalledWith(
+            defaultOptions.textTemplate
+          );
+          expect(resultAction.loading).toBe(false);
+        });
+
+        it('should log an error in the console when the promise is rejected', async () => {
+          const mockCopyToClipboard = quanticUtils.copyToClipboard;
+          const element = createTestComponent();
+
+          const resultAction = element.shadowRoot.querySelector(
+            selectors.quanticResultAction
+          );
+
+          // @ts-ignore
+          mockCopyToClipboard.mockImplementationOnce(() =>
+            Promise.reject(new Error('Copy to clipboard error'))
+          );
+
+          element.dispatchEvent(copyToClipboardEvent);
+          await flushPromises();
+
+          expect(console.error).toHaveBeenCalledWith(
+            'Copy to clipboard action failed.',
+            expect.any(Error)
+          );
+          expect(resultAction.loading).toBe(false);
+        });
+
+        describe('when the tooltip is displayed', () => {
+          beforeAll(() => {
+            jest.useFakeTimers();
+          });
+
+          afterAll(() => {
+            jest.useRealTimers();
+          });
+
+          it('should display the success label then reset the initial label after 1000ms', async () => {
+            const element = createTestComponent();
+            await flushPromisesWithoutTimeout();
+
+            const resultAction = element.shadowRoot.querySelector(
+              selectors.quanticResultAction
+            );
+            expect(resultAction).not.toBeNull();
+
+            element.dispatchEvent(copyToClipboardEvent);
+            await flushPromisesWithoutTimeout();
+
+            expect(resultAction.label).toBe(defaultOptions.successLabel);
+
+            jest.advanceTimersByTime(1000);
+            await flushPromisesWithoutTimeout();
```

Comment:
```suggestion
            await jest.advanceTimersByTime(1000);
```

Reply:
changed it to `advanceTimersByTimeAsync`, but it was still passing with the await for the `advanceTimersByTime `